terraform/deployments/ga4-analytics/project_iam_custom_roles.tf

```
@@ -0,0 +1,144 @@
+resource "google_project_iam_custom_role" "roles--gds_bigquery_read_access" {
+  description = "Created on: 2023-10-27"
+  permissions = [
+    "bigquery.datasets.get",
+    "bigquery.tables.get",
+    "bigquery.tables.getData"
+  ]
+  role_id = "GDS_BQ_read_access"
+  title   = "GDS BQ read access"
+}
+
+resource "google_project_iam_custom_role" "roles--gds_bigquery_saved_query_writer" {
+  description = "Created on: 2023-11-23"
+  permissions = [
+    "bigquery.savedqueries.create",
+    "bigquery.savedqueries.delete",
+    "bigquery.savedqueries.update",
+    "dataform.repositories.create",
+    "dataform.repositories.get",
+    "dataform.repositories.list"
+  ]
+  role_id = "GDS_BQ_saved_query_writer"
+  title   = "GDS BQ saved query writer"
+}
+
+resource "google_project_iam_custom_role" "roles--gds_logging_alerts_writer" {
+  description = "Logger access to create alerts"
+  permissions = [
+    "logging.logEntries.create",
+    "logging.logEntries.list",
+    "logging.logMetrics.create",
+    "logging.logMetrics.delete",
+    "logging.logMetrics.get",
+    "logging.logMetrics.list",
+    "logging.logMetrics.update",
+    "logging.notificationRules.create",
+    "logging.notificationRules.get",
+    "logging.notificationRules.list",
+    "logging.notificationRules.update",
+    "monitoring.alertPolicies.create",
+    "monitoring.alertPolicies.delete",
+    "monitoring.alertPolicies.get",
+    "monitoring.alertPolicies.list",
+    "monitoring.alertPolicies.update"
+  ]
+  role_id = "GDS_log_alert_writer"
+  title   = "GDS log alert writer"
+}
+
+resource "google_project_iam_custom_role" "roles--gds_bigquery_user" {
+  description = "Adds transfers update to the standard BigQuery User role"
+  permissions = [
+    "bigquery.bireservations.get",
+    "bigquery.capacityCommitments.get",
+    "bigquery.capacityCommitments.list",
+    "bigquery.datasets.get",
+    "bigquery.datasets.getIamPolicy",
+    "bigquery.jobs.create",
+    "bigquery.jobs.get",
+    "bigquery.jobs.list",
+    "bigquery.jobs.listAll",
+    "bigquery.jobs.listExecutionMetadata",
+    "bigquery.models.list",
+    "bigquery.readsessions.create",
+    "bigquery.readsessions.getData",
+    "bigquery.reservationAssignments.list",
+    "bigquery.reservationAssignments.search",
+    "bigquery.reservations.get",
+    "bigquery.reservations.list",
+    "bigquery.savedqueries.get",
+    "bigquery.savedqueries.list",
+    "bigquery.tables.get",
+    "bigquery.tables.getData",
+    "bigquery.tables.list",
+    "bigquery.transfers.get",
+    "bigquery.transfers.update",
+    "bigquerymigration.translation.translate",
+    "resourcemanager.projects.get",
+  ]
+  role_id = "gds.bigquery.user"
+  title   = "GDS BQ user"
+}
+
+resource "google_project_iam_custom_role" "roles--gds_bigquery_editor" {
```

```suggestion
resource "google_project_iam_custom_role" "gds_bigquery_editor" {
```

Nit: don't think the prefix adds anything to the name